Use C10_UNUSED instead of (void)X

r-barnes opened the pull request:
Summary:
Auto-generated with
```
buck run //scripts/rbarnes/regex_multiline_replacer:regex_multiline_replacer -- --find '^(\s*for\s*\()(const.*\n)\s*\(void\)[A-Za-z]+;\s*//\s*Suppress.*\s*\n(.*)'  --replace '\1C10_UNUSED \2\3' `find caffe2/ -regex ".*\.\(cpp\|h\)"`
```

Differential Revision: D33432600


cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

cyyever commented:
Some loops have two C10_UNUSED

Skylion007 commented on aten/src/ATen/native/layer_norm.cpp:
```diff
@@ -256,7 +256,7 @@ std::tuple<Tensor, Tensor, Tensor> math_native_layer_norm(
     stat_shape.push_back(input_shape[idx]);
   }
-  for (const auto idx C10_UNUSED : c10::irange(axis, input.dim())) {
+  for (C10_UNUSED const auto idx C10_UNUSED : c10::irange(axis, input.dim())) {
```
Ugh, it went a bit haywire on these loops...

cyyever commented:
The changes from " [[maybe_unused]]" to "C10_UNUSED" may be unnecessary because we may want to change them back and remove  C10_UNUSED  once C++17 gets full support in all platforms.

r-barnes committed:
Use C10_UNUSED instead of (void)X (#137239)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/137239

Auto-generated with
```
buck run //scripts/rbarnes/regex_multiline_replacer:regex_multiline_replacer -- --find '^(\s*for\s*\()(const.*\n)\s*\(void\)[A-Za-z]+;\s*//\s*Suppress.*\s*\n(.*)'  --replace '\1C10_UNUSED \2\3' `find caffe2/ -regex ".*\.\(cpp\|h\)"`
```

Test Plan: Sandcastle

Differential Revision: D33432600
